src/Entity/EntityId.php

```diff
@@ -30,6 +55,91 @@ public function getSerialization() {
 	}
 
 	/**
+	 * Returns an array with 3 elements: the foreign repository name as the first element, the local ID as the last
+	 * element and everything that is in between as the second element.
+	 *
+	 * EntityId::joinSerialization can be used to restore the original serialization from the parts returned.
+	 *
+	 * @param string $serialization
+	 * @return string[] Array containing the serialization split into 3 parts.
+	 */
+	public static function splitSerialization( $serialization ) {
+		self::assertValidSerialization( $serialization );
+
+		$parts = explode( ':', self::normalizeIdSerialization( $serialization ) );
+		$localPart = array_pop( $parts );
+		$repoName = array_shift( $parts ) ?: '';
+		$prefixRemainder = implode( ':', $parts );
+
+		return array( $repoName, $prefixRemainder, $localPart );
+	}
+
+	/**
+	 * Builds an ID serialization from the parts returned by EntityId::splitSerialization.
+	 *
+	 * @param string[] $parts
+	 * @return string
+	 */
+	public static function joinSerialization( array $parts ) {
+		return implode( ':', array_filter( $parts ) );
```

Comment:
This is not fully compatible with the `splitSerialization` implementation above. `splitSerialization` guarantees the first element is never empty. But this code here returns the exact same result for all these test cases:
* `[ 'Q1' , '', '' ]`
* `[ '' , 'Q1', '' ]`
* `[ '' , '', 'Q1' ]`

This is quite bad. Two of these cases should fail (I believe the first two).

This code also fails for `[ 'wd', '', '0' ]`, `[ '0', '0', '0' ]` and so on, because the string `"0"` is falsy in PHP. But according to the regex on top this string is a valid ID.